src/cc_catalog_airflow/dags/provider_api_scripts/metropolitan_museum_of_art.py

```diff
@@ -0,0 +1,257 @@
+        logger.error('No retries remaining.  Failure.')
+        raise Exception('Retries exceeded')
+
+    response = delayed_requester.get(
+        endpoint,
+        params=query_params,
+        timeout=60
+    )
+    if response is not None and response.status_code == 200:
+        try:
+            response_json = response.json()
+        except Exception as e:
+            logger.warning(f'Could not get response_json.\n{e}')
+            response_json = None
+
+    if response_json is None:
+        logger.warning(f'Bad response_json:  {response_json}')
+        logger.warning(
+            'Retrying:\n_get_response_json(\n'
+            f'    {endpoint},\n'
+            f'    {query_params},\n'
+            f'    retries={retries - 1}'
+            ')'
+        )
+        response_json = _get_response_json(
+            query_params,
+            endpoint=endpoint,
+            retries=retries - 1
+        )
+
+    return response_json
+
+
+def _extract_the_data(object_ids):
+    for i in object_ids:
+        _get_data_for_each_image(i)
+
+
+def _get_data_for_each_image(object_id):
+
+    endpoint = (
+        'https://collectionapi.metmuseum.org/public/collection/v1/objects/{}'
+        .format(object_id)
+    )
+
+    object_json = _get_response_json(None, endpoint)
+
+    if object_json is None:
+        logger.error('Unable to process object ID : {}'.format(object_id))
+        return None
+
+    message = object_json.get('message')
+    if message:
+        logger.warning(f'{message} : {object_id}')
+        return None
+
+    isCC0 = object_json.get('isPublicDomain')
+    if isCC0 is None or isCC0 is False:
+        logger.warning('CC0 license not detected')
+        return None
+
+    foreign_url = object_json.get('objectURL')
+    if foreign_url is None:
+        logger.warning(f'No landing page detected for: {object_id}')
+        return None
+
+    image_info = object_json.get('primaryImage')
+    if image_info is None:
+        logger.warning(f'No image found for {object_id}')
+        return None
+
+    title = object_json.get('title')
+    creator_name = object_json.get('artistDisplayName')
+    foreign_id = object_id
+    meta_data = _create_meta_data(object_json)
+    image_url = image_info
+
+    thumbnail = ''
+    if '/original/' in image_url:
+        thumbnail = image_url.replace('/original/', '/web-large/')
+
+    other_images = object_json.get('additionalImages', None)
+
+    if other_images is not None and len(other_images) > 1:
+        extra_image_index = 1
+        meta_data['set'] = foreign_url
+
+    image_data = {
+        'foreign_landing_url': foreign_url,
+        'image_url': image_url,
+        'thumbnail_url': thumbnail,
+        'foreign_identifier': foreign_id,
+        'creator': creator_name,
+        'title': title,
+        'meta_data': meta_data
+    }
+
+    _process_image_data(image_data)
+
+    if other_images is not None and len(other_images) > 1:
+        for image in other_images:
```

Comment:
Note that `image` is the alias of one of your imports, and you're redefining it here.  This is a bit confusing, so I suggest choosing a different name for either that import, or this variable.